contracts/HubPool.sol

```diff
@@ -270,19 +287,17 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         // Decrement the unclaimedPoolRebalanceLeafCount.
         refundRequest.unclaimedPoolRebalanceLeafCount--;
 
-        // Transfer the bondAmount to back to the proposer, if this was not done before for this refund bundle.
-        if (!refundRequest.proposerBondRepaid) {
-            refundRequest.proposerBondRepaid = true;
+        // Transfer the bondAmount to back to the proposer, if this the last executed leaf. Only sending this once all
+        // leafs have been executed acts to force the data worker to execute all bundles or they wont receive their bond.
+        if (refundRequest.unclaimedPoolRebalanceLeafCount == 0)
             bondToken.safeTransfer(refundRequest.proposer, bondAmount);
-        }
 
-        // TODO call into canonical bridge to send PoolRebalance.netSendAmount for the associated
-        // PoolRebalance.tokenAddresses, to the target PoolRebalance.chainId. this will likely happen within a
-        // x_Messenger contract for each chain. these messengers will be registered in a separate process that will follow
-        // in a later PR.
+        _sendTokensToTargetChain(poolRebalance, refundRequestId);
+        _executeRelayerRefundOnTargetChain(poolRebalance, refundRequestId);
```

Comment:
all the logic of sending tokens to the chains and sending the L1->L2 calls is wrapped into these two functions :)

Reply:
On L2, what happens if the relayer refund method on the spoke pool is executed before tokens arrive there?

Reply:
it'll revert as there is not enough money to pay everyone.

Reply:
Let me be sure I have the right understanding here:
1. This will just send over a verified _root_ that will be saved on the L2. No funds get automatically distributed.
2. If funds haven't arrived _and someone tried to distribute on the L2_, their transaction will revert.

Is my understanding correct? In no situation should the L2 side of the `_executeRelayerRefundOnTargetChain` transaction revert, right?

Reply:
this is exactly right @mrice32